src/python/pants/goal/run_tracker.py

```diff
@@ -304,8 +304,9 @@ def store_stats(self):
     # Upload to remote stats db.
     stats_url = self.get_options().stats_upload_url
     if stats_url:
-      t = threading.Thread(target=self.post_stats, args=(stats_url, stats, self.get_options().stats_upload_timeout))
-      t.start()
+      pid = os.fork()
+      if pid == 0:
+        self.post_stats(stats_url, stats, timeout=self.get_options().stats_upload_timeout)
```

Comment:
It's more that without the try finally, the child could potentially throw an exception that would trigger other behavior we don't want.